closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -693,33 +704,31 @@ find_crypto_library(jboolean traceEnabled, const char *chomepath)
             fprintf(stdout, "Attempting to load library bundled with JDK from: %s\n", libPath);
         }
 
-        for (i = 0; i < numOfLibs; i++) {
-            size_t file_len = strlen(libNames[i]);
-            /* Allocate memory for the new file name with the path. */
-            char *libNameWithPath = (char *)malloc(path_len + file_len + 1);
-
-            if (NULL == libNameWithPath) {
-                if (traceEnabled) {
-                    fprintf(stderr, "\tFailed to allocate memory for file name with path.\n");
-                }
-                continue;
-            }
+        size_t file_len = strlen(bundledLibName);
+        /* Allocate memory for the new file name with the path. */
+        char *libNameWithPath = (char *)malloc(path_len + file_len + 1);
 
-            strcpy(libNameWithPath, libPath);
-            strcat(libNameWithPath, libNames[i]);
-
-            /* Load OpenSSL Crypto library bundled with JDK. */
+        if (NULL == libNameWithPath) {
             if (traceEnabled) {
-                fprintf(stdout, "\tAttempting to load: %s\n", libNames[i]);
+                fprintf(stderr, "\tFailed to allocate memory for file name with path.\n");
             }
-            result = load_crypto_library(traceEnabled, libNameWithPath);
+            free(libPath);
+            return NULL;
+        }
 
-            free(libNameWithPath);
+        strcpy(libNameWithPath, libPath);
+        strcat(libNameWithPath, bundledLibName);
```

Comment:
Just one allocation is needed; line 689 should become
```c
        size_t path_len = strlen(chomepath) + sizeof(pathSuffix) - 1 + sizeof(bundledLibName) - 1;
```
and this should be:
```c
        strcat(libPath, bundledLibName);
```
Then line 728 should be removed.